### ISSUE TITLE
Intermittent `saveas` error

### ISSUE THREAD

**Issue by samuela** [plotly/plotly.py#575]
I'm using `saveas` to save a lot of frames in an animated sequence, and occasional I get this very peculiar error:

```
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-11-cea42e8543cd> in <module>()
----> 1 rotaty_mcroteface(conn, 6011, 1, 11)

<ipython-input-10-289b04e5962c> in rotaty_mcroteface(conn, exp_id, dim_x, dim_y, n_frames, zoom_factor, **kwds)
     26                         'center': {'x': 0, 'y': 0, 'z': 0}})))
     27 
---> 28         plotly.plotly.image.save_as(fig, 'frames/stuff_{}_frame{:04d}.png'.format(exp_id, i))

/Users/samuelainsworth/research/res-venv/lib/python2.7/site-packages/plotly/plotly/plotly.pyc in save_as(cls, figure_or_data, filename, format, width, height, scale)
    785             filename += '.' + format
    786 
--> 787         img = cls.get(figure_or_data, format, width, height, scale)
    788 
    789         f = open(filename, 'wb')

/Users/samuelainsworth/research/res-venv/lib/python2.7/site-packages/plotly/plotly/plotly.pyc in get(figure_or_data, format, width, height, scale)
    708                                              "from plotly could "
    709                                              "not be translated.")
--> 710             raise exceptions.PlotlyError(return_data['error'])
    711 
    712     @classmethod

KeyError: 'error'
```

which doesn't really tell me much about what's wrong here.

**Comment by cldougl** [comment]
Agreed that error message isn't very helpful, it's related to API throttling limits. Limits can be found on the [plans page](https://plot.ly/products/cloud/).
Closing as it's very related to: https://github.com/plotly/plotly.py/issues/447 and further discussion can take place there.

**Comment by samuela** [comment]
I was actually using plotly in offline mode. Do API throttling limits affect offline plotly as well?

**Comment by cldougl** [comment]
@samuela you'd have to specify how you were using `saveas` or attach a code snippet? 
Typically using `saveas()` is only available in online mode: https://plot.ly/python/static-image-export/#export-static-image-online. See the exporting image documentation for an offline export example: https://plot.ly/python/static-image-export/#export-static-image-offline

**Comment by samuela** [comment]
The code snippet looks like:

```python
surf = <... build surface ...>
    
for i in range(n_frames):
    fig = go.Figure(
        data=[surf],
        layout=go.Layout(
            width=800,
            height=600,
            scene=go.Scene(
                camera={
                    'eye': {
                        'x': np.sin(float(i) / n_frames * np.pi * 2) * np.sqrt(2) * zoom_factor,
                        'y': np.cos(float(i) / n_frames * np.pi * 2) * np.sqrt(2) * zoom_factor,
                        'z': zoom_factor / 2
                    },
                    'up': {'x': 0, 'y': 0, 'z': 1},
                    'center': {'x': 0, 'y': 0, 'z': 0}})))

    plotly.plotly.image.save_as(fig, 'frames/frame{:04d}.png'.format(i))
```

**Comment by jonmmease** [comment]
Update: Offline programmatic static image export support has been released in 3.2.0: See https://medium.com/@plotlygraphs/plotly-py-end-of-summer-updates-5422c98b9058
